core/src/main/scala/org/genivi/sota/core/campaigns/CampaignLauncher.scala

```diff
@@ -35,8 +41,25 @@ object CampaignLauncher {
     Future.successful(devices.map(_ -> Set(pkg.id)).toMap)
   }
 
-  def launch (deviceRegistry: DeviceRegistry, updateService: UpdateService, id: Campaign.Id, lc: LaunchCampaign)
-             (implicit db: Database, system: ActorSystem, ec: ExecutionContext): Future[List[Uuid]] = {
+  def launch (deviceRegistry: DeviceRegistry, updateService: UpdateService, id: Campaign.Id, lc: LaunchCampaign,
+              messageBus: MessageBusPublisher)
+             (implicit db: Database, system: ActorSystem, ec: ExecutionContext, mat: ActorMaterializer)
+      : Future[List[Uuid]] = {
+    def sendMsg(namespace: Namespace, devices: Set[Uuid], pkgId: PackageId): Future[Unit] = {
+      import org.genivi.sota.marshalling.CirceMarshallingSupport._
+      implicit val _config = system.settings.config
+
+      //check that this update involves a treehub image
+      if(pkgId.name.get.startsWith("treehub-")) {
+        for {
+          pkg <- db.run(Packages.byId(namespace, pkgId))
+          response <- new PackageStorage().retrieveResponse(pkg)
```

Comment:
This just returns a `HttpResponse` that the client will follow to find the real file.